File: orig/man/echo.m
{ Copyright (C) 1981 by Bell Laboratories, Inc., and Whitesmiths Ltd. }
.NM echo echo arguments to output
.SY
.UL "echo [ argument ... ]"
.FU
.UL echo
copies its command line arguments to its output as a line
of text with one space between each argument.
If there are no arguments, no output is produced.
.EG
To see if your system is alive:
.Q1
echo hello world!
.S "hello world!"
.Q2

File: orig/filters/echo.cc
#include "../p2c.h"
#include "filters.h"

// Copyright (C) 1981 by Bell Laboratories, Inc., and Whitesmiths Ltd.
// echo -- echo command line arguments to output
void echo()
{
    integer i, j;
    string argstr;

    i = 1;
    while (getarg(i, argstr, MAXSTR)) {
        if (i > 1)
            putc(BLANK);
        for (j = 1; j <= length(argstr); ++j)
            putc(argstr[j]);
        i = i + 1;
    }
    if (i > 1)
        putc(NEWLINE);
}
